src/docfx/lib/log/Telemetry.cs

```diff
@@ -17,26 +17,42 @@ internal static class Telemetry
     {
         private static readonly TelemetryClient s_telemetryClient = new TelemetryClient(TelemetryConfiguration.CreateDefault());
 
-        private static readonly Metric s_operationTimeMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"time", "name", "os", "version", "repo", "branch"));
-        private static readonly Metric s_errorCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"error", "code", "level", "os", "version", "repo", "branch"));
-        private static readonly Metric s_cacheCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"cache", "name", "state", "os", "version", "repo", "branch"));
-        private static readonly Metric s_buildItemCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"count", "name", "os", "version", "repo", "branch"));
+        private static readonly Metric s_operationTimeMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"Time", "Name", "OS", "Version", "Repo", "Branch", "CorrelationId"));
+        private static readonly Metric s_errorCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"BuildLog", "Code", "Level", "Type", "OS", "Version", "Repo", "Branch", "CorrelationId"));
+        private static readonly Metric s_cacheCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"Cache", "Name", "State", "OS", "Version", "Repo", "Branch", "CorrelationId"));
+        private static readonly Metric s_buildItemCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"Count", "Name", "OS", "Version", "Repo", "Branch", "CorrelationId"));
 
         private static readonly string s_version = typeof(Telemetry).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "<null>";
         private static readonly string s_os = RuntimeInformation.OSDescription ?? "<null>";
 
         private static string s_repo = "<null>";
         private static string s_branch = "<null>";
 
+        private static Dictionary<string, string> s_eventDimensions = new Dictionary<string, string>();
```

Comment:
There was a potential bug here for multiple docsets, since telemetry config, repo, branch is tied to a docset in docfx, consider use `AsyncLocal<T>` for `s_repo`, `s_branch`, `s_telemetryConfig` as `AsyncLocal<T>` so each docset can have different values.

Reply:
are we now building multiple docset in parallel?

Reply:
why is it tied to docset? should it be the same between docsets?

---
In reply to: [365082539](https://github.com/dotnet/docfx/pull/5433#discussion_r365082539) [](ancestors = 365082539)

Reply:
I see we are running docset build parallelly. Please ignore my last question.